packages/fast-usdc-contract/src/exos/advancer.ts

```diff
@@ -201,8 +207,22 @@ export const prepareAdvancerKit = (
             const { EUD } = decoded.query;
             log(`decoded EUD: ${EUD}`);
             assert.typeof(EUD, 'string');
-            // throws if the bech32 prefix is not found
+
+            // throws if it's neither CAIP-10 nor bare bech32.
             const destination = chainHub.resolveAccountId(EUD);
+            const destChain = chainOfAccount(destination);
+            const info = await when(chainHub.getChainInfoByChainId(destChain));
+
+            // These are the cases that are currently supported
+            if (
+              destChain !== settlementAddress.chainId &&
```

Comment:
`settlementAddress` is a CosmosChainAddress so this comparison with a CAIP-2 doesn't make sense. I think here it's operating as `if (true &&`. What was the intent? Consider a test test that would detect this bug.